pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -886,6 +889,16 @@ func proposeBatch(
 		//      slice of entries. See etcd-io/raft#57.
 		maybeDeductFlowTokens(ctx, p.flowControlHandle(ctx), handles, ents)
 	}
+
+	// If the context on the proposal is set up for tracing, register the
+	// proposal with rafttrace. This will add the trace to updates.
+	for i, e := range ents {
+		if s := tracing.SpanFromContext(props[i].Context()); s != nil && s.RecordingType() == tracingpb.RecordingVerbose {
+			p.registerForTracing(props[i], e)
+			// NB: Only trace the first entry in the batch.
```

Comment:
Why only the first entry? The code as of now allows multiple proposals to be traced (up to `maxSize`). Consider breaking the loop only if the `maxSize` is exceeded (e.g. the `registerForTracing` call can return a bool indicating that).

Reply:
The idea here was that the entries would likely have a similar lifecycle and tracing one would be sufficient. I could alternatively trace the last one, but I wasn't sure which was better. I didn't want to use up the full "tracing budget" with the one proposal that came in.

Reply:
Note that I changed to the last entry as all entries are required to complete and the last is going to be the "last" one. We can look at adjusting this in the future if its a problem.